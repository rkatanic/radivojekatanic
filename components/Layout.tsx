import Menu from "./Menu";

interface Props {
  children: JSX.Element;
}

const Layout = ({ children }: Props): JSX.Element => (
  <div className="z-10 flex min-h-screen flex-col items-start justify-center bg-white  dark:bg-neutral-900">
    <Menu />
    <div className="h-60 w-full bg-gradient-to-r from-lime-100 via-sky-100 to-blue-100"></div>
    <div className="m-auto w-full max-w-3xl">{children}</div>
  </div>
);

export default Layout;
