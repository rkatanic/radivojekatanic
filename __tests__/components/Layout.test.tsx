import Layout from "@/components/Layout";
import { render } from "@testing-library/react";

jest.mock("next/config", () => () => ({
  publicRuntimeConfig: {
    lastModifiedDate: "01-10-2020",
  },
}));
const useRouter = jest.spyOn(require("next/router"), "useRouter");

describe("Layout", (): void => {
  it("should render", (): void => {
    (useRouter as jest.Mock).mockReturnValue({ asPath: "/", pathname: "/" });
    const { baseElement } = render(
      <Layout>
        <div>children</div>
      </Layout>
    );

    expect(baseElement).toMatchSnapshot();
  });
});
