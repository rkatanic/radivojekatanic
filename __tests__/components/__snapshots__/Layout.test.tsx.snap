// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should render 1`] = `
<body>
  <div>
    <div
      class="relative mx-auto flex min-h-screen flex-col justify-center gap-8 px-6 text-slate-900 dark:bg-neutral-900 dark:text-neutral-200"
    >
      <div
        class="mx-auto flex w-full max-w-2xl items-baseline justify-between"
      >
        <div
          class="flex gap-6 pb-8 pt-16 font-medium"
        >
          <a
            href="/"
          >
            About
          </a>
        </div>
        <div
          class="flex items-center gap-2"
        >
          <button
            class="group relative flex h-9 w-9 items-center justify-center rounded-xl border border-slate-300 px-2 text-sm text-slate-500 shadow-sm transition hover:text-slate-800 dark:border-neutral-800 dark:bg-neutral-800 dark:text-neutral-500 dark:hover:bg-neutral-700/50 dark:hover:text-neutral-200"
          >
            <div
              class="invisible absolute right-[calc(100%+12px)] translate-y-1 rounded-lg p-1 px-2 text-sm font-medium opacity-0 transition-[transform,opacity] before:absolute before:-right-[6px] before:top-1/2 before:h-0 before:w-0 before:-translate-y-1/2 before:border-b-[7px] before:border-l-[7px] before:border-t-[7px] before:border-b-transparent before:border-t-transparent group-hover:visible group-hover:translate-y-0 group-hover:opacity-100 bg-slate-800 text-slate-50 before:border-l-slate-800 dark:bg-neutral-200 dark:text-neutral-900 dark:before:border-l-neutral-200"
            >
              theme
            </div>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              version="1.1"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8 0c-4.418 0-8 3.582-8 8s3.582 8 8 8 8-3.582 8-8-3.582-8-8-8zM2 8c0-3.314 2.686-6 6-6v12c-3.314 0-6-2.686-6-6z"
              />
            </svg>
          </button>
          <div
            class="relative cursor-pointer select-none"
          >
            <div
              class="group flex h-9 items-center gap-2 rounded-xl border border-slate-300 p-1 px-3 text-sm text-slate-600 shadow-sm hover:text-slate-800 dark:border-neutral-800 dark:bg-neutral-800 dark:text-neutral-400 dark:hover:bg-neutral-700/50 dark:hover:text-neutral-200"
            >
              English
              <div
                class="flex h-5 flex-col"
              >
                <svg
                  class="pointer-events-none dark:stroke-neutral-500 dark:group-hover:stroke-neutral-400"
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <polyline
                    points="18 15 12 9 6 15"
                  />
                </svg>
                <svg
                  class="pointer-events-none -mt-0.5 dark:stroke-neutral-500 dark:group-hover:stroke-neutral-400"
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <polyline
                    points="6 9 12 15 18 9"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="mx-auto flex w-full max-w-2xl items-center justify-between pb-8"
      >
        <div
          class="flex items-center gap-4"
        >
          <img
            alt="my-photo"
            src="/me.png"
          />
          <div
            class="flex flex-col gap-2"
          >
            <h3
              class="text-xl font-semibold"
            >
              fullName
            </h3>
            <p
              class="font-medium text-slate-700 dark:text-neutral-300"
            >
              role
            </p>
            <div
              class="flex items-center gap-4 text-slate-600 dark:text-neutral-400"
            >
              <a
                class="group flex cursor-pointer items-center gap-2 hover:text-slate-900 hover:underline dark:hover:text-neutral-200"
                href="mailto:rkatanic@outlook.com"
              >
                <svg
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M4 4h16c1.1 0 2 .9 2 2v12c0 1.1-.9 2-2 2H4c-1.1 0-2-.9-2-2V6c0-1.1.9-2 2-2z"
                  />
                  <polyline
                    points="22,6 12,13 2,6"
                  />
                </svg>
                rkatanic@outlook.com
              </a>
              <div
                class="h-4 w-px bg-slate-200 dark:bg-neutral-700"
              />
              <a
                class="group flex cursor-pointer items-center gap-2 hover:text-slate-900 hover:underline dark:hover:text-neutral-200"
                href="https://github.com/rkatanic"
                rel="noreferrer noopener"
                target="_blank"
              >
                <svg
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M9 19c-5 1.5-5-2.5-7-3m14 6v-3.87a3.37 3.37 0 0 0-.94-2.61c3.14-.35 6.44-1.54 6.44-7A5.44 5.44 0 0 0 20 4.77 5.07 5.07 0 0 0 19.91 1S18.73.65 16 2.48a13.38 13.38 0 0 0-7 0C6.27.65 5.09 1 5.09 1A5.07 5.07 0 0 0 5 4.77a5.44 5.44 0 0 0-1.5 3.78c0 5.42 3.3 6.61 6.44 7A3.37 3.37 0 0 0 9 18.13V22"
                  />
                </svg>
                github
              </a>
              <div
                class="h-4 w-px bg-slate-200 dark:bg-neutral-700"
              />
              <a
                class="group flex cursor-pointer items-center gap-2 hover:text-slate-900 hover:underline dark:hover:text-neutral-200"
                href="https://linkedin.com/in/radivoje-katanic"
                rel="noreferrer noopener"
                target="_blank"
              >
                <svg
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M16 8a6 6 0 0 1 6 6v7h-4v-7a2 2 0 0 0-2-2 2 2 0 0 0-2 2v7h-4v-7a6 6 0 0 1 6-6z"
                  />
                  <rect
                    height="12"
                    width="4"
                    x="2"
                    y="9"
                  />
                  <circle
                    cx="4"
                    cy="4"
                    r="2"
                  />
                </svg>
                linkedin
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        class="mx-auto w-full max-w-2xl flex-1 pb-16"
      >
        <div>
          children
        </div>
      </div>
    </div>
  </div>
</body>
`;
